storage/constraints.go

```diff
@@ -27,14 +27,28 @@ type Constraints struct {
 	Count uint64
 }
 
+// Directives describes a set of storage directives.
+type Directives struct {
```

Comment:
There's no need to duplicate the Constraints struct and parsing code below.
Leave it as constraints and we can separately rename the struct in a separate refactoring PR if we decide to do that

Reply:
there's quite a number of dependencies using Constraints, and im not too sure if we want to refactor those fields to a pointer in this PR hence I made another Directives struct with the relevant pointer fields.

Reply:
hi ian, we use Directives in the ParseDirectives func as well which is used in the cmd. 

https://github.com/juju/juju/blob/a03d1de59b2697b3a164e4c9997ef2cad4d73b12/storage/constraints.go#L131